kubernetes-deployment/bootstrap.py

```diff
@@ -395,9 +397,33 @@ def remove_nodes(cluster_config, node_list_config):
 
 
 
+def maintain_expert(cluster_config, maintain_config, node_config, job_name):
+
+    module_name = "maintainlib.{0}".format(job_name)
+    module = importlib.import_module(module_name)
+
+    job_class = getattr(module, job_name)
+    job_instance = job_class(cluster_config, node_config, False)
+
+    job_instance.run()
+
+
+
+def maintain_nodes(cluster_config, node_list_config, job_name):
+
+    # Todo: load maintain from a DB such as etcd instead of a yaml file.
+    maintain_config = load_yaml_file("maintain.yaml")
+
+    for host in node_list_config['machinelist']:
+
+        maintain_expert(cluster_config, maintain_config, node_list_config['machinelist'][host], job_name)
+
+
+
 def option_validation(args):
```

Comment:
>option_validation [](start = 4, length = 17)

given that we have CI in place, can you add unit test for this function?

Reply:
Add unitest for validation?

Reply:
yes, add unit test to see if validation can catch it when user submit with wrong flag 

---
In reply to: [164982731](https://github.com/Microsoft/pai/pull/281#discussion_r164982731) [](ancestors = 164982731)

Reply:
Next pr will merge them into one. Do ut in the future.